forge/ee/routes/tables/index.js

```diff
@@ -0,0 +1,303 @@
+                        type: 'object',
+                        properties: {
+                            name: { type: 'string' },
+                            schema: { type: 'string' }
+                        }
+                    }
+                },
+                '4xx': {
+                    $ref: 'APIError'
+                },
+                500: {
+                    $ref: 'APIError'
+                }
+            }
+        }
+    }, async (request, reply) => {
+        // paginate the list of tables
+        const tables = await app.tables.getTables(request.team, request.params.databaseId)
+        if (!tables) {
+            return reply.status(404).send({ code: 'not_found', error: 'Database not found' })
+        }
+        reply.send(tables)
+    })
+
+    /**
+     * @name /api/v1/teams/:teamId/databases/:databaseId/tables/:tableName
+     * @description Get a specific table schema in a team database
+     * @static
+     * @memberof forge.routes.api.team.tables
+     */
+    app.get('/:databaseId/tables/:tableName', {
+        preHandler: app.needsPermission('team:database:list'),
+        schema: {
+            summary: '',
+            tags: ['FF tables'],
+            params: {
+                type: 'object',
+                properties: {
+                    databaseId: { type: 'string' },
+                    tableName: { type: 'string' }
+                }
+            },
+            response: {
+                200: {
+                    $ref: 'DatabaseTable'
+                },
+                '4xx': {
+                    $ref: 'APIError'
+                },
+                500: {
+                    $ref: 'APIError'
+                }
+            }
+        }
+    }, async (request, reply) => {
+        // get the table schema
+        const table = await app.tables.getTable(request.team, request.params.databaseId, request.params.tableName)
+        if (!table) {
+            return reply.status(404).send({ error: 'Table not found' })
+        }
+        reply.send(table)
+    })
+
+    /**
+     * @name /api/v1/teams/:teamId/databases/:databaseId/tables/:tableName/data
+     * @description Get sample data from a specific table in a team database
+     * @static
+     * @memberof forge.routes.api.team.tables
+     */
+    app.get('/:databaseId/tables/:tableName/data', {
+        preHandler: app.needsPermission('team:database:list'),
+        schema: {
+            summary: '',
+            tags: ['FF tables'],
+            params: {
+                type: 'object',
+                properties: {
+                    databaseId: { type: 'string' },
+                    tableName: { type: 'string' }
+                }
+            },
+            response: {
+                200: {
+                    type: 'array',
+                    items: {
+                        type: 'object',
+                        additionalProperties: true // allow any properties in the table rows
+                    }
```

Comment:
Although this route is designed to grab a sample, we may wish to include meta or even pagination at some future point.
Changing this to an object now avoids any future duck type checking awkwardness